File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/chain_responsibility/BossManager.java
package jv.com.tencent.cain.design_pattern.chain_responsibility;

/**
 * @author : jiangyu
 * @date : 2020/8/8
 * @desc : xxx
 */
public class BossManager extends AbstractManager {
    public BossManager(String name) {
        super(name);
    }

    @Override
    void handleRequest(Request request) {
        if (request.getRequestType() == Request.REQUEST_TYPE_VACATION) {
            if (request.getRequestCount() <= 7) {
                System.out.println(request.getRequestContent() + ", " + request.getRequestCount() + ", " + name + "已批准");
            } else {
                System.out.println(request.getRequestContent() + ", " + request.getRequestCount() + ", " + name + "已拒绝");
            }
        } else if (request.getRequestType() == Request.REQUEST_TYPE_RAISE_SALARY) {
            if (request.getRequestCount() <= 1500) {
                System.out.println(request.getRequestContent() + ", " + request.getRequestCount() + ", " + name + "已批准");
            } else {
                System.out.println(request.getRequestContent() + ", " + request.getRequestCount() + ", " + name + "已拒绝");
            }
        }
    }
}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/chain_responsibility/BossManager.kt
package kt.com.tencent.cain.desigin_pattern.chain_responsibility

import kt.com.tencent.cain.desigin_pattern.chain_responsibility.Request.Companion.REQUEST_TYPE_RAISE_SALARY
import kt.com.tencent.cain.desigin_pattern.chain_responsibility.Request.Companion.REQUEST_TYPE_VACATION

/**
 * @author : jiangyu
 * @date   : 2020/8/8
 * @desc   : xxx
 */
class BossManager(name: String) : Manager(name) {

    override fun handleRequest(request: Request) {
        when (request.requestType) {
            REQUEST_TYPE_VACATION -> {
                when {
                    request.requestCount <= 7 -> {
                        println("${request.requestContent}, ${request.requestCount}, ${name}已批准")
                    }
                    else -> {
                        println("${request.requestContent}, ${request.requestCount}, ${name}已拒绝")

                    }
                }
            }
            REQUEST_TYPE_RAISE_SALARY -> {
                when {
                    request.requestCount <= 1500 -> {
                        println("${request.requestContent}, ${request.requestCount}, ${name}已批准")
                    }
                    else -> {
                        println("${request.requestContent}, ${request.requestCount}, ${name}已拒绝")
                    }
                }
            }
        }
    }
}